integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPortForward.java

```diff
@@ -0,0 +1,439 @@
+  private static String portForwardFileNameProfix = RESULTS_ROOT + "/port-foward";
+  private static String hostName = "localhost";
+
+  /**
+   * Install operator, create two custom images using model in image with model files
+   * and create two one cluster domains, one for testing forwarding a local port to
+   * admin default port(non-secure and secure port) and another one for testing
+   * forwarding a local port to administration port.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(3) List<String> namespaces) {
+    logger = getLogger();
+
+    // get a unique operator namespace
+    logger.info("Get a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a unique domain namespace for adminDefaultPortDomain
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    adminDefaultPortDomainNamespace = namespaces.get(1);
+
+    // get a unique domain namespace for administrationPortDomain
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    administrationPortDomainNamespace = namespaces.get(2);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace,
+        adminDefaultPortDomainNamespace, administrationPortDomainNamespace);
+
+    // create and verify WebLogic domain image for adminDefaultPortDomain
+    // using model in image with model files
+    String imageName = createAndVerifyDomainImage(adminDefaultPortDomainNamespace,
+        PORTFORWARD_ADMIN_DEFAULT_IMAGE_NAME, PORTFORWARD_ADMIN_DEFAULT_MODEL_FILE);
+
+    // create and verify the domain is created and ready to use
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain(imageName, adminDefaultPortDomainNamespace, adminDefaultPortDomainUid);
+
+    // create and verify WebLogic domain image for administrationPortDomain
+    // using model in image with model files
+    imageName = createAndVerifyDomainImage(administrationPortDomainNamespace,
+        PORTFORWARD_ADMINISTRATION_IMAGE_NAME, PORTFORWARD_ADMINISTRATION_MODEL_FILE);
+
+    // create and verify the domain is created and ready to use
+    createAndVerifyDomain(imageName, administrationPortDomainNamespace, administrationPortDomainUid);
+  }
+
+  /**
+   * kill port-forward process.
+   */
+  @AfterAll
+  void tearDown() {
+    stopPortForwardProcess();
+  }
+
+  /**
+   * The test that `kubectl port-foward` is able to forward a local port to default channel port
+   * (7001 in this test).
+   * Verify that the WLS admin console can be accessed using http://localhost:localPort/console/login/LoginForm.jsp
+   */
+  @Order(1)
```

Comment:
Do we need explicit Order annotation ? Test can be run independently.

Reply:
yes we do because of dependencies of test cases.

Reply:
Tests can be made independent  if we re-org the tests method(s) as follow ..
testPortForwardAdministrationPort() {

  stopPortForwardProcess()
  startPortForwardProcess( 9002 ) 
  verifyAdminConsoleAccessible( true ) 
  startPortForwardProcess( 7001 )
  verifyAdminConsoleAccessible( false )
  startPortForwardProcess( 7002 )
  verifyAdminConsoleAccessible( false )
  } 

A single tests passed after I made the change locally.

Reply:
The purpose of using tag "Order" is to make use of the setting of previous test to save overall running time. Sure I can stop/start portfoward. The trade off will be the longer test running time. Is there any reasons we can't use Order?

Reply:
It is a good practice to make the test method independent of each other, so that individual test can be run independently. There was a discussion regarding this in a k8s standup.  start(stop)PortForwardProcess does not take much time. The most time taking job to configure/start two domains is taken care by initAll() method.

Reply:
done